### PR TITLE
Amend num state

### PR DESCRIPTION
**Pull request by Kavikick**
Takes care of the edge numbers such as 0 and 9, as well as change the behavior of the num state to only set the state once the number has been accepted. This pull request hinges on pull request #83

### CODE REVIEW

**Comment by jgfoster** [comment]
Now that #83 is merged, you may want to see if you can reduce the apparent changes.

**Review by prestoncarman — APPROVED** [review]
LGTM

Nice to simplify this.

File: test/NumberCollectorState.cpp [hunk 1]
```diff
@@ -5,32 +5,36 @@
 
 #include "TankControllerLib.h"
 
+unittest(Edge_Case) {
```
Comment (jgfoster) [inline test/NumberCollectorState.cpp:8]: Thank you for adding a test for the broken code. It demonstrates good discipline!

**Review by jgfoster — APPROVED** [review]
LGTM

**Pull request merged**